nitro-protocol/contracts/libraries/signature-logic/StrictTurnTaking.sol

```diff
@@ -11,21 +11,28 @@ library StrictTurnTaking {
      * @notice Require supplied arguments to comply with turn taking logic, i.e. each participant signed the one state, they were mover for.
      * @dev Require supplied arguments to comply with turn taking logic, i.e. each participant signed the one state, they were mover for.
      * @param fixedPart Data describing properties of the state channel that do not change with state updates.
-     * @param recoveredVariableParts An ordered array of structs, each struct describing dynamic properties of the state channel and must be signed by corresponding moving participant.
+     * @param proof Array of recovered variable parts which constitutes a support proof for the candidate. The proof is a validation for the supplied candidate.
+     * @param candidate Recovered variable part the proof was supplied for. The candidate state is supported by proof states.
      */
     function requireValidTurnTaking(
         INitroTypes.FixedPart memory fixedPart,
-        INitroTypes.RecoveredVariablePart[] memory recoveredVariableParts
+        INitroTypes.RecoveredVariablePart[] memory proof,
+        INitroTypes.RecoveredVariablePart memory candidate
     ) internal pure {
-        _requireValidInput(fixedPart.participants.length, recoveredVariableParts.length);
+        _requireValidInput(fixedPart.participants.length, proof.length + 1);
```

Comment:
It's a bit surprising to pass `proof.length + 1`. I suggest we just remove the check on the proof length. Having `candidate` as a separate parameter is giving us this check for free implicitly.

Reply:
Do you suggest we remove checks on proof length entirely (from `_requireValidInput` method)?
I see no other simple way, which can back up this check.

Reply:
That is what I was suggesting yes. However, now I am not so sure :-P . For now I'm happy to leave this as is, actually.

Reply:
Nevertheless, I am still sure `proof.length + 1` is a bad practise, so small refactor will be implemented 😃